great_expectations/rule_based_profiler/parameter_builder/parameter_container.py

```diff
@@ -251,7 +251,7 @@ def _build_parameter_node_tree_for_one_parameter(
 
 def get_parameter_value_by_fully_qualified_parameter_name(
     fully_qualified_parameter_name: str,
-    domain: Domain,
+    domain: Optional[Domain] = None,
```

Comment:
Why is this optional now?

Reply:
@talagluck This is required for the very reason you mentioned before: the method `get_parameter_value_by_fully_qualified_parameter_name()` should be useful to retrieve values from `variables` -- we use this in the `DomainBuilder` classes, where, by definition, the `Domain` does not yet exist (it is being built).  I suspect that at some point we will gather a sufficient amount of justification to introduce a proper `RuleExecutionState` class, which will handle a number of these cases inside (but feel that we can defer it for now).